src/addDefaultProps.coffee

```diff
@@ -0,0 +1,21 @@
+import {isFunction} from './util/helpers'
+import addProps from './addProps'
+
+addDefaultProps = (createDefaults) ->
+  addProps (props) ->
+    defaults = if isFunction createDefaults
+      createDefaults props
+    else
+      createDefaults
+
+    Object.keys(defaults).reduce(
```

Comment:
I'd probably go with a less functional implementation "under the hood" here especially since in Coffeescript iterating on an object is nicer

So maybe:
```
newProps = {...props}

(newProps[key] = val) for own key, val in defaults when not props[key]?

newProps # implicit return
```